modules/system/assets/js/october.parser.js

```diff
@@ -0,0 +1,353 @@
+                    result += ":";
+                    type = ":";
+                } else if (type === ":") {
+                    var body = getBody(str, i);
+
+                    i = i + body.originLength - 1;
+                    result += parse(body.body);
+
+                    type = "afterBody";
+                } else if (type === "afterBody" || type === "needKey") {
+                    var last = i;
+                    while (str[last] === "," || isBlankChar(str[last])) {
+                        last++;
+                    }
+                    if (str[last] === "}" && last === str.length - 1) {
+                        while (result[result.length - 1] === ",") {
+                            result = result.substr(0, result.length - 1);
+                        }
+                        result += "}";
+                        return result;
+                    } else if (last !== i && result !== "{") {
+                        result += ",";
+                        type = "needKey";
+                        i = last - 1;
+                    }
+                }
+            }
+            throw new Error("Broken JSON object near " + result);
+        }
+
+        /*
+         * array
+         */
+        if (str[0] === "[") {
+            var result = "[";
+            var type = "needBody";
+            for (var i = 1; i < str.length; i++) {
+                if (" " === str[i] || "\n" === str[i] || "\t" === str[i]) {
+                    continue;
+                } else if (type === "needBody") {
+                    if (str[i] === ",") {
+                        result += "null,";
+                        continue;
+                    }
+                    if (str[i] === "]" && i === str.length - 1) {
+                        if (result[result.length - 1] === ",") result = result.substr(0, result.length - 1);
+                        result += "]";
+                        return result;
+                    }
+
+                    var body = getBody(str, i);
+
+                    i = i + body.originLength - 1;
+                    result += parse(body.body);
+
+                    type = "afterBody";
+                } else if (type === "afterBody") {
+                    if (str[i] === ",") {
+                        result += ",";
+                        type = "needBody";
+
+                        // deal with mistake ","
+                        while (str[i + 1] === "," || isBlankChar(str[i + 1])) {
+                            if (str[i + 1] === ",") result += "null,";
+                            i++;
+                        }
+                    } else if (str[i] === "]" && i === str.length - 1) {
+                        result += "]";
+                        return result;
+                    }
+                }
+            }
+            throw new Error("Broken JSON array near " + result);
+        }
+    }
+
+    var g = (typeof exports === "undefined") ? (window.$.oc.JSON = {}) : exports;
```

Comment:
@ayumihamsaki I think this part is incorrect. If would like to support `everything` then the proper way to use eigther:
```javascript
// AMD support
    if (typeof define === 'function' && define.amd) {
        define(function () { return MyModule; });
    // CommonJS and Node.js module support.
    } else if (typeof exports !== 'undefined') {
        // Support Node.js specific `module.exports` (which can be a function)
        if (typeof module !== 'undefined' && module.exports) {
            exports = module.exports = MyModule;
        }
        // But always support CommonJS module 1.1.1 spec (`exports` cannot be a function)
        exports.MyModule = MyModule;
    } else {
        global.MyModule = MyModule;
    }
```

OCMS way:
https://github.com/octobercms/october/blob/fb2aa1730cc08e8563908b54b70e942bd2988f5e/modules/system/assets/js/framework.extras.js#L242

Reply:
You confused me and helped me out at the same time 😄

Reply:
Just returning export is incorrect, and not used anymore.